gap/bliss.gi

```diff
@@ -12,141 +13,247 @@
-    if not (ForAll(list, IsDuplicateFreeList) and Union(list) = colors) then
-      ErrorNoReturn("Digraphs: DigraphCanonicalLabelling: usage,\n",
-                    "the union of the lists in the second arg should equal ",
-                    "[1 .. ", DigraphNrVertices(graph), "],");
-    fi;
-
-    for i in [1 .. Length(list)] do
-      colors{list[i]} := [1 .. Length(list[i])] * 0 + i;
-    od;
-  else
-    if not (Length(list) = DigraphNrVertices(graph)
-            and ForAll(list, c -> IsPosInt(c) and 1 <= c
-                                  and c <= DigraphNrVertices(graph))) then
-      ErrorNoReturn("Digraphs: DigraphCanonicalLabelling: usage,\n",
-                    "the second arg must be a list of length ",
-                    DigraphNrVertices(graph), " of integers in [1 .. ",
-                    DigraphNrVertices(graph), "],");
-    fi;
-    colors := list;
+  n := DigraphNrVertices(digraph);
+  colours := DIGRAPHS_ValidateVertexPartition(n, list);
+  if colours = fail then
+    ErrorNoReturn("Digraphs: DigraphCanonicalLabelling: usage,\n",
+                  "the argument <list> does not define a valid colouring of ",
+                  "the vertices of\n<digraph>. Either <list> should be a list ",
+                  "of length <n> consisting of integers\nin the range ",
+                  "[0 .. <n>], or <list> should be a list of non-empty ",
+                  "disjoint\nlists whose union is the vertices of <digraph>,");
   fi;
-
-  return DIGRAPH_CANONICAL_LABELING_COLORS(graph, colors);
+  return DIGRAPH_CANONICAL_LABELING_COLORS(digraph, colours);
 end);
 
 #
 
 InstallMethod(IsIsomorphicDigraph, "for digraphs",
 [IsDigraph, IsDigraph],
-function(g1, g2)
+function(gr1, gr2)
+  local act;
+
+  if gr1 = gr2 then
+    return true;
+  fi;
 
   # check some invariants
-  if DigraphNrVertices(g1) <> DigraphNrVertices(g2)
-      or DigraphNrEdges(g1) <> DigraphNrEdges(g2)
-      or IsMultiDigraph(g1) <> IsMultiDigraph(g2) then
+  if DigraphNrVertices(gr1) <> DigraphNrVertices(gr2)
+      or DigraphNrEdges(gr1) <> DigraphNrEdges(gr2)
+      or IsMultiDigraph(gr1) <> IsMultiDigraph(gr2) then
     return false;
   fi; #JDM more!
 
-  if IsMultiDigraph(g1) then
-    return OnMultiDigraphs(g1, DigraphCanonicalLabelling(g1))
-      = OnMultiDigraphs(g2, DigraphCanonicalLabelling(g2));
+  if IsMultiDigraph(gr1) then
+    act := OnMultiDigraphs;
+  else
+    act := OnDigraphs;
   fi;
-  return OnDigraphs(g1, DigraphCanonicalLabelling(g1))
-    = OnDigraphs(g2, DigraphCanonicalLabelling(g2));
+
+  return act(gr1, DigraphCanonicalLabelling(gr1))
+    = act(gr2, DigraphCanonicalLabelling(gr2));
+end);
+
+#
+
+InstallMethod(IsIsomorphicDigraph, "for digraphs and homogeneous lists",
+[IsDigraph, IsDigraph, IsHomogeneousList, IsHomogeneousList],
+function(gr1, gr2, c1, c2)
+  local m, colour1, n, colour2, class_sizes, i;
+
+  if IsMultiDigraph(gr1) or IsMultiDigraph(gr2) then
+    ErrorNoReturn("Digraphs: IsIsomorphicDigraph: usage,\n",
```

Comment:
Why is this not supported? On the face of it, isn't this just the same as the non-coloured case?